tests/test_transforms/test_timestamp/test_event_transform.py

```diff
@@ -0,0 +1,164 @@
+
+
+@pytest.fixture
+def ts_check_pipeline_with_event_transform(random_seed) -> TSDataset:
+    periods_df = 30
+    periods_df_exoc = periods_df + 10
+    df = pd.DataFrame({"timestamp": pd.date_range("2020-01-01", periods=periods_df)})
+    df["segment"] = ["segment_1"] * periods_df
+    df["target"] = np.arange(periods_df)
+
+    df_exoc = pd.DataFrame({"timestamp": pd.date_range("2020-01-01", periods=periods_df_exoc)})
+    df_exoc["segment"] = ["segment_1"] * periods_df_exoc
+    df_exoc["holiday"] = np.random.choice([0, 1], size=periods_df_exoc)
+
+    df = TSDataset.to_dataset(df)
+    df_exoc = TSDataset.to_dataset(df_exoc)
+    tsds = TSDataset(df, freq="D", df_exog=df_exoc, known_future="all")
+    return tsds
+
+
+@pytest.fixture
+def ts_check_event_transform_expected() -> TSDataset:
+    periods = 10
+    dataframes = [pd.DataFrame({"timestamp": pd.date_range("2020-01-01", periods=periods)}) for _ in range(3)]
+    for i in range(len(dataframes)):
+        df = dataframes[i]
+        df["segment"] = [f"segment_{i}"] * periods
+        df["target"] = np.arange(periods)
+        if i == 0:
+            df["holiday"] = np.array([0, 1, 0, 1, 1, 0, 0, 0, 1, 0])
+            df["holiday_binary_prev"] = np.array([1, 0, 1, 0, 0, 0, 1, 1, 0, 0])
+            df["holiday_binary_post"] = np.array([0, 0, 1, 0, 0, 1, 1, 0, 0, 1])
+            df["holiday_distance_prev"] = np.array([1, 0, 1, 0, 0, 0, 0.5, 1, 0, 0])
+            df["holiday_distance_post"] = np.array([0, 0, 1, 0, 0, 1, 0.5, 0, 0, 1])
+        elif i == 1:
+            df["holiday"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            df["holiday_binary_prev"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            df["holiday_binary_post"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            df["holiday_distance_prev"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            df["holiday_distance_post"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+        else:
+            df["holiday"] = np.array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1])
+            df["holiday_binary_prev"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            df["holiday_binary_post"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            df["holiday_distance_prev"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            df["holiday_distance_post"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+    df = pd.concat(dataframes)
+    df = TSDataset.to_dataset(df)
+    tsds = TSDataset(df, freq="D", known_future="all")
+    return tsds
+
+
+@pytest.fixture
+def ts_check_event_transform() -> TSDataset:
+    periods = 10
+    dataframes = [pd.DataFrame({"timestamp": pd.date_range("2020-01-01", periods=periods)}) for _ in range(3)]
+    for i in range(len(dataframes)):
+        df = dataframes[i]
+        df["segment"] = [f"segment_{i}"] * periods
+        df["target"] = np.arange(periods)
+        if i == 0:
+            df["holiday"] = np.array([0, 1, 0, 1, 1, 0, 0, 0, 1, 0])
```

Comment:
As I said exog variables should be in separate dataframe and should be passed into `df_exog` parameter